benchmarks/src/gatling/scala/org/apache/polaris/benchmarks/simulations/CreateTreeDataset.scala

```diff
@@ -64,11 +67,31 @@ class CreateTreeDataset extends Simulation {
   private val createdViews = new AtomicInteger()
 
   // --------------------------------------------------------------------------------
-  // Workload: Authenticate and store the access token for later use
+  // Authentication related workloads:
+  // * Authenticate and store the access token for later use every minute
+  // * Wait for an OAuth token to be available
+  // * Stop the token refresh loop
   // --------------------------------------------------------------------------------
-  val authenticate: ScenarioBuilder = scenario("Authenticate using the OAuth2 REST API endpoint")
-    .feed(authenticationActions.feeder())
-    .exec(authenticationActions.authenticateAndSaveAccessToken)
+  val continuouslyRefreshOauthToken: ScenarioBuilder =
+    scenario("Authenticate every minute using the Iceberg REST API")
+      .asLongAs(_ => shouldRefreshToken.get())(
+        feed(authenticationActions.feeder())
+          .exec(authenticationActions.authenticateAndSaveAccessToken)
+          .pause(1.minute)
```

Comment:
nit: Could we make this configurable too?

Reply:
Well, we could but it would be a tricky parameter.

This action is equivalent to an uninterruptible sleep.  So in the worse case, with this 1 minute value, Gatling will continue running for nothing during 60 seconds after the last response from all other queries has been received.

If we make this parameter, I can imagine how people would think that a token needs be refreshed only every 30 minutes.  And now they have to wait up to 30 minutes before the process actually completes.

I will likely give another try at this element in the future.  Like, instead of a 1 minute pause, run a loop of 60 "sleep 1s" actions with the same check `_ => shouldRefreshToken.get()` that is done earlier.  Or something similar.

Reply:
I agree this is misleading, as you could easily infer that changing to `.pause(30.minutes)` wouldn't have any consequence on the test duration.

Suggestion (but I don't know Gatling very well): instead of continuously polling the atomic integer until it becomes false, couldn't the caller send an interrupt signal to the scenario?

But it's OK imo to investigate improvements later.

Reply:
Thanks for the clarification! I'm ok with improving this later.